### ISSUE TITLE
export to wav broken

### ISSUE THREAD

**Issue by unifirer** [erew123/alltalk_tts#71]
[diagnostics.log](https://github.com/erew123/alltalk_tts/files/13933880/diagnostics.log)

nothing happens when i click export to wav with split to 100

**To Reproduce**
Steps to reproduce the behaviour:
i put in a text of 900k characters, and waited until all the 4200 chunks at size 4 were completed and clicked on export

**Screenshots**
![Screenshot (32)](https://github.com/erew123/alltalk_tts/assets/152607209/f7ea5625-81f0-4c7f-9515-ac660d29e919)

all the console code was fine, looked like this, this is the last part after i clicked export

[AllTalk TTSGen] 5.87 seconds. LowVRAM: False DeepSpeed: True
[AllTalk TTSGen] Link’s laughter was carried upon the wind. “Then let’s go.” He whirled about. heel tipping off the edge of the plateau as he gave them all a wide grin. “We’ve got a future to build.” He let himself freefall for a moment before tapping his Slate and vanishing into tendrils of blue upon Nayru’s winds.
[AllTalk TTSGen] 11.44 seconds. LowVRAM: False DeepSpeed: True

AllTalk was updated: v1.9
Custom Python environment: [default}
Text-generation-webUI was updated: latest 7th jan 2024

i checked in outputs folder, nothing more than 7mbs, and ive checked task manager, no process taking more than 1% CPU and ive waited more than 30mins and clicked a few more times just to be sure after 10mins

**Comment by erew123** [comment]
BTW, when it exports to wav file, there should be a popup rotating disk at the top right hand corner of the screen (the same as when it is generating). If you only have a small amount of text to export, say 50 or so, this may be little more than a flash at the top of the screen as the export happens pretty quickly. When I exported 1000+ wavs, this took 1 minute+ to perform, so the spinning disk was there for that duration. So you should have a status update when its actually working correctly.

**Comment by Suiyou** [comment]
Sorry to butt in but I have the same issue. I used Chrome and Opera:

![03](https://github.com/erew123/alltalk_tts/assets/4419512/e02b5dc7-46ae-4147-b418-0d100421ff48)
![01](https://github.com/erew123/alltalk_tts/assets/4419512/44731635-9da9-4d13-96ab-f97783818498)

In my case I just use a script to join the files directly in the terminal to get around this.

**Comment by erew123** [comment]
@Suiyou Thanks for the screenshot. Everything that matters is happening within the webpage, so that's where we are invesitgating here. 

**If you have time** would you do me a favour. I'm wondering if this is something changed in a Python module somewhere as to how its serving web pages OR if its something to do with it seeing the esm on the javascript name of crunker. As I don't have the issue myself, its a bit hard to diagnose which one of the two could be causing it, or if its actually something else stripping things e.g. antivirus or similar stripping information out.

Would you be able to drop me a diagnostics log file and I can compare my systems python modules and update/downgrade as needed so I can discount that possibility. https://github.com/erew123/alltalk_tts?tab=readme-ov-file#-how-to-make-a-diagnostics-report-file

**Comment by Suiyou** [comment]
Here you go: 
[diagnostics.log](https://github.com/erew123/alltalk_tts/files/13940762/diagnostics.log)

**Comment by erew123** [comment]
@Suiyou Thanks for that! 

@Suiyou @unifirer Whilst I cant directly replicate the issue, what I've done is set it to specifically serve JS files with a mimetype of application/javascript. This is an update to tts_server.py, so please `git pull` an update and try again (You shouldnt need to clear your browser cache, but do at least close all instances of TTS Generator before opening a new one up). 

In short, when you open the TTS Generator in your browser, if you press F12 and select "Console" it should show no errors....

![image](https://github.com/erew123/alltalk_tts/assets/35898566/de9c54b5-f90c-49eb-9d40-aeb3ee858b24)

I'm going to optimistically (famous last words) flag this as closed, so let me know if there is a further issues specifically with Exporting Wav files. 

Thanks

**Comment by unifirer** [comment]
> Hi @unifirer
> 
> This has me baffled for a few minutes as there have been no-code updates to this for a week+, double-checked github, downloaded the current file again to be sure etc.
> 
> I've tested and I have no problems exporting or any other functions.
> 
> Have you a million to one chance opened the file directly from disk e.g. in your URL on your web-browser, does it say something like `c:\text-generation-webui\extensions\alltalk_tts\templates\.....`
> 
> ![image](https://private-user-images.githubusercontent.com/35898566/296725430-14a8102a-f96a-4f84-a27a-3e934dc332a4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUzOTIzMjksIm5iZiI6MTcwNTM5MjAyOSwicGF0aCI6Ii8zNTg5ODU2Ni8yOTY3MjU0MzAtMTRhODEwMmEtZjk2YS00Zjg0LWEyN2EtM2U5MzRkYzMzMmE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE2VDA4MDAyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM5ZDE1YmFlZGEwNjhiYTY5OTk1N2FjYWM2ZGM5MzU5ZDhhMzNkNDA5YzAzNzNlMTExMjMxMWU1MGQyZWZmMjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.JHRfGdhoU0v4E1fZsuLhIBwvtyCP0Q7O3lP6CiiwBfs)
> 
> Do you have it opened as the URL running from AllTalk's webserver e.g. `http://127.0.0.1:7851/static/tts_chunk_player/tts_chunk_player.html`
> 
> ![image](https://private-user-images.githubusercontent.com/35898566/296725798-71032430-c406-49a8-a424-a89d7de3707e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUzOTIzMjksIm5iZiI6MTcwNTM5MjAyOSwicGF0aCI6Ii8zNTg5ODU2Ni8yOTY3MjU3OTgtNzEwMzI0MzAtYzQwNi00OWE4LWE0MjQtYTg5ZDdkZTM3MDdlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE2VDA4MDAyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQzOTI1MjYwYWYyZGMzYjM3MmQxYWFkOTJhYzc1YjhmY2I1NDU4YmVkYjA3ZWE3YmQ3Mzg1NDdmOTQzY2JjZmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.I2UG9EYKeFt-W7P-FrKJikl6W5KxYnnKMALUjgc-XKU)
> 
> The reason I ask is that running it straight of the disk drive would cause it to think there was cross site scripting when it tries to export the wav. To put that simply, you would be running from `c:\text-generation-webui\........ etc` and when the page calls to export the wav, it would be calling for other files from `http://127.0.0.1:785/static/.... etc` hence cross site (2x different locations).
> 
> In that situation your web-browser says no-no that's naughty you're trying to do something you shouldn't, so Ill silently block it happening. However, if you pressed F12 within your web-browser and open the developer console, you would see something along the following lines
> 
> ![image](https://private-user-images.githubusercontent.com/35898566/296727481-154dbf0f-6675-49bf-aaa1-32384f29b9b4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUzOTIzMjksIm5iZiI6MTcwNTM5MjAyOSwicGF0aCI6Ii8zNTg5ODU2Ni8yOTY3Mjc0ODEtMTU0ZGJmMGYtNjY3NS00OWJmLWFhYTEtMzIzODRmMjliOWI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE2VDA4MDAyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA4NjQwNjcxYWQ0ZGYxOGIwMGJmMmE0YmQ0NzQyZDkzZjE4ZWY3ZThiMzY0N2Q3ZjE4YjhlODg2MmI0OTVkNjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.BD3h7mnxUaq5rVGVpeaONa0EBrz-n_nKC99bkkzDdQA)
> 
> Could that be what's happened for you?
> 
> If not and you are connecting via the `http://127.0.0.1:7851/static/tts_chunk_player/tts_chunk_player.html` would you be so kind as to tell me what browser you are using and perhaps press F12 when trying to export and see if there is a `console` message I could look at.
> 
> Let me know
> 
> Thanks

thank you very much for your reply, ive just been moving because theres a housing shortage in wellington and its my first time trying it but i always keep it open because im doing this a lot

this is always open now
http://127.0.0.1:7851/static/tts_chunk_player/tts_chunk_player.html 

im using firefox, ill try again on opera, it might actually be firefox, a lot of problems on it thats not on opera, thank you for the suggestion,
the wheel doesnt spin when i click export. 
ive closed it and will report back tmr 
as i spent 4hrs cooking today haha. beef spare ribs are delicious in the oven for 4hrs at 160C fyi, use any spice, and lots. 

another thing i noticed, using chunk size 4, the processing time doubled compared to the old way of a huge text. it might fix audio problems tho. havnt listened to it yet but the old way always had problems for long text.

thank you very much for ur hardwork
